vulkaninfo/vulkaninfo.c

```diff
@@ -4716,7 +4716,7 @@ static void AppDumpExtensions(const char *indent, const char *layer_name, const
         printf("\tcount = %d\n", extension_count);
     }
 
-    const bool is_device_type = strcmp(layer_name, "Device") == 0;
+    const bool is_device_type = layer_name && (strcmp(layer_name, "Device") == 0);
```

Comment:
strcmp could be switched to strncmp with the limit as VK_MAX_EXTENSION_NAME_SIZE.

Reply:
Hmm the variable name is bit bad.
Anyway the function seems to assume the pointer is valid; it is never called with `NULL` and it would not make sense if it was.

Perhaps there shoud just be `assert(layer_name && strlen(layer_name) > 0)`.

Reply:
The first time layer_name is used in that function, it's checked if layer_name is valid, but yes it never seems to get called with NULL, so maybe an assert is more suitable.

Reply:
So this make me wonder if the calling code is always correct, since I can't infer the intent here,

below is some of the calling code, why does it pass "Layer-Device" instead of the layer_name?

        char *layer_name = inst.global_layers[i].layer_properties.layerName;
            AppGetPhysicalDeviceLayerExtensions(&gpus[j], layer_name, &count, &props);
            if (html_output) {
                AppDumpExtensions("\t\t\t", "Layer-Device", count, props, out);
            } else if (human_readable_output) {
                AppDumpExtensions("\t\t", "Layer-Device", count, props, out);
            }

Reply:
Hence I commented the name of the variable is misleading.

The `vulkaninfo` output looks like this:

```
VK_LAYER_LUNARG_core_validation (LunarG Validation Layer) Vulkan version 1.1.101, layer version 1
        Layer Extensions        count = 1
                VK_EXT_debug_report                 : extension revision  6
        Devices         count = 1
                GPU id       : 0 (AMD Radeon R9 200 Series)
                Layer-Device Extensions count = 2
                        VK_EXT_debug_marker                 : extension revision  4
                        VK_EXT_validation_cache             : extension revision  1
```

I guess it would look weird if the layer name was repeated there instead of only "Layer-Device".